### PR TITLE
Updated rename_file method in zk_nn_client to use multiop and added test

### PR DESCRIPTION
**Pull request by prb2**
Passes test case and was able to do copyFromLocal and cat.

Can @pelmers and/or @nichhk code review this?

### CODE REVIEW

File: zookeeper/source/zk_nn_client.cc [hunk 1]
```diff
@@ -664,6 +664,7 @@ namespace zkclient{
```
Comment (nichhk) [inline zookeeper/source/zk_nn_client.cc:664]: Why doesn't this function take error_code as a param?

File: zookeeper/source/zk_nn_client.cc [hunk 2]
```diff
@@ -664,6 +664,7 @@ namespace zkclient{
 
 		int error_code = 0;
 		auto data = std::vector<std::uint8_t>();
+		auto ops = std::vector<std::shared_ptr<ZooOp>>();
 
 		std::string src_znode = "/fileSystem" + src;
```
Comment (nichhk) [inline zookeeper/source/zk_nn_client.cc]: please use ZooKeeperPath()

File: test/namenode/NameNodeTest.cc [hunk 3]
```diff
@@ -91,6 +91,31 @@ namespace {
...
+		ASSERT_EQ(0, error_code);
+		ASSERT_EQ("File data", std::string(new_file_data.begin(), new_file_data.end()));
+		LOG(INFO) << "New: " << std::string(new_file_data.begin(), new_file_data.end());
+
+		auto new_block_data = std::vector<std::uint8_t>();
+		zk->get("/fileSystem/new_name/block-0000000000", new_block_data, error_code);
+		ASSERT_EQ(0, error_code);
+		ASSERT_EQ("Block uuid", std::string(new_block_data.begin(), new_block_data.end()));
+		LOG(INFO) << "New: " << std::string(new_block_data.begin(), new_block_data.end());
+
```
Comment (nichhk) [inline test/namenode/NameNodeTest.cc:117]: Can you put some asserts to make sure that old_name doesn't exist anymore?

File: zookeeper/source/zk_nn_client.cc [hunk 4]
```diff
@@ -859,47 +849,47 @@ namespace zkclient{
...
+		auto data = std::vector<std::uint8_t>();
+		if (zk->get(HEALTH_BACKSLASH + data_node + STATS, data, error_code)) {
+			LOG(ERROR) << CLASS_NAME << "Getting data node stats failed with " << error_code;
+		}
+
+		zkclient::DataNodePayload * payload = (zkclient::DataNodePayload *) (&data[0]);
+
+		DatanodeIDProto* id = dn_info->mutable_id();
+		id->set_ipaddr(split_address[0]);
+		id->set_hostname("localhost"); // TODO: Fill out with the proper value
```
Comment (nichhk) [inline zookeeper/source/zk_nn_client.cc:872]: Do you know what this TODO is?

**Review by nichhk — CHANGES_REQUESTED** [review]
Looks good to me. Just some small things~~~

File: test/namenode/NameNodeTest.cc (hunk 3 above)
Comment (prb2) [reply test/namenode/NameNodeTest.cc:117]: Done.

d7aabdd3727499bfe042bde05de9d9b9a5a7d041

File: zookeeper/source/zk_nn_client.cc (hunk 4 above)
Comment (prb2) [reply zookeeper/source/zk_nn_client.cc:872]: No idea...not sure who wrote this.

File: zookeeper/source/zk_nn_client.cc (hunk 2 above)
Comment (prb2) [reply zookeeper/source/zk_nn_client.cc]: Done.

d7aabdd3727499bfe042bde05de9d9b9a5a7d041

File: zookeeper/source/zk_nn_client.cc (hunk 1 above)
Comment (prb2) [reply zookeeper/source/zk_nn_client.cc:664]: As we discussed, it now prints out errors when something in multiop fails.

2e990fc

**Comment by prb2** [comment]
@nichhk Ready to merge?

**Pull request merged**